src/spawn.rs

```diff
@@ -1,166 +1,63 @@
-use std::future::Future;
-
-#[cfg(feature = "async_std")]
-pub use async_std_impl::*;
-#[cfg(feature = "smol")]
-pub use smol_impl::*;
+/// Spawns the given actor into the tokio runtime, returning an [`Address`] to it.
+///
+/// The 2nd parameter is the mailbox size with `None` creating an unbounded mailbox.
 #[cfg(feature = "tokio")]
-pub use tokio_impl::*;
-#[cfg(feature = "wasm_bindgen")]
-pub use wasm_bindgen_impl::*;
+#[cfg_attr(docsrs, doc(cfg(feature = "tokio")))]
+pub fn spawn_tokio<A>(actor: A, mailbox_size: Option<usize>) -> crate::Address<A>
+where
+    A: crate::Actor<Stop = ()>,
+{
+    let (address, context) = crate::Context::new(mailbox_size);
+
+    tokio::spawn(context.run(actor));
 
-/// An `Spawner` represents anything that can spawn a future to be run in the background. This is
-/// used to spawn actors.
-pub trait Spawner {
-    /// Spawn the given future.
-    fn spawn<F: Future<Output = ()> + Send + 'static>(&mut self, fut: F);
+    address
 }
 
+/// Spawns the given actor into the async_std runtime, returning an [`Address`] to it.
+///
+/// The 2nd parameter is the mailbox size with `None` creating an unbounded mailbox.
 #[cfg(feature = "async_std")]
-mod async_std_impl {
-    use super::*;
-    use crate::{Actor, ActorManager, Address};
+#[cfg_attr(docsrs, doc(cfg(feature = "async_std")))]
+pub fn spawn_async_std<A>(actor: A, mailbox_size: Option<usize>) -> crate::Address<A>
```

Comment:
I theory, these functions could be generated by a macro but it is only a handful of functions so I didn't want to bother :)